addons/mail/tests/test_mail_channel_as_guest.py

```diff
@@ -4,13 +4,35 @@
 import odoo
 from odoo.tests import HttpCase
 
+
 @odoo.tests.tagged('-at_install', 'post_install')
-class TestMailGuestPages(HttpCase):
-    def test_mail_channel_as_guest(self):
-        """Checks that the invite page redirects to the channel and that all
-        modules load correctly on the welcome and channel page"""
-        channel = self.env['mail.channel'].create({
+class TestMailPublicPage(HttpCase):
+    """Checks that the invite page redirects to the channel and that all
+    modules load correctly on the welcome and channel page when authenticated as various users"""
+
+    def setUp(self):
+        super().setUp()
+        self.channel = self.env['mail.channel'].create({
             'name': 'Test channel',
             'public': 'public',
         })
-        self.start_tour(f"/chat/{channel.id}/{channel.uuid}", "mail/static/tests/tours/mail_channel_as_guest_tour.js")
+
+    def open_channel_page_as_user(self, user):
+        route, tour = f"/chat/{self.channel.id}/{self.channel.uuid}",  "mail/static/tests/tours/mail_channel_as_guest_tour.js"
+        self.authenticate(user, user)
+        self.start_tour(route, tour)
+        # Second run of the tour as the first call has side effects, like creating user settings or adding members to
+        # the channel, so we need to run it again to test different parts of the code.
+        self.start_tour(route, tour)
+
+    def test_mail_channel_public_page_as_guest(self):
+        self.open_channel_page_as_user('')
+
+    def test_mail_channel_public_page_as_portal(self):
+        self.open_channel_page_as_user('portal')
+
+    def test_mail_channel_public_page_as_internal(self):
+        self.open_channel_page_as_user('demo')
+
+    def test_mail_channel_public_page_as_admin(self):
```

Comment:
Sort them alphabetically or add number in the names, otherwise the tests don't run in the same order as they are defined which is sometimes confusing.